core/src/main/java/org/elasticsearch/common/geo/builders/PointListBuilder.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.common.geo.builders;
+
+import com.vividsolutions.jts.geom.Coordinate;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * A builder for a list of points (of {@link Coordinate} type).
+ * Enables chaining of individual points either as long/lat pairs
+ * or as {@link Coordinate} elements, arrays or collections.
+ */
+public class PointListBuilder {
```

Comment:
Super nit: Can we refactor this class name to CoordinateListBuilder? That way the naming is a bit more consistent, such as:
```
CoordinateListBuilder cl = new CoordinateListBuilder();
int size = in.readVInt();
for (int i=0; i < size; i++) {
  cl.coordinate(readCoordinateFrom(in));
}
```